### ISSUE TITLE
Can't scrape some metrics Windows VM

### ISSUE THREAD

**Issue by caijian76** [kubevirt/kubevirt#9801]
**What happened**:
kubevirt_vmi_memory_used_bytes , kubevirt_vmi_memory_unused_bytes ,kubevirt_vmi_memory_usable_bytes ,can only scrape metrics Linux VM , Windows VM can't show any data ?

How do I get windows memory usage?

Kubevirt Version: 0.59
K8S version: 1.25.7
Runtime :  containerd

**Comment by aburdenthehand** [comment]
Hello, thank you for raising this bug.
Just to confirm, did you install the guest-agent for Windows on this VM?

**Comment by caijian76** [comment]
yes, agent has installed发自我的iPhone------------------ Original ------------------From: aburdenthehand  ***@***.***>Date: Wed,May 31,2023 10:14 PMTo: kubevirt/kubevirt ***@***.***>Cc: caijian76 ***@***.***>, Author ***@***.***>Subject: Re: [kubevirt/kubevirt] Can't scrape some metrics Windows VM (Issue#9801)
Hello, thank you for raising this bug.
Just to confirm, did you install the guest-agent for Windows on this VM?

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: ***@***.***>
[
{
***@***.***": "http://schema.org",
***@***.***": "EmailMessage",
"potentialAction": {
***@***.***": "ViewAction",
"target": "https://github.com/kubevirt/kubevirt/issues/9801#issuecomment-1570321772",
"url": "https://github.com/kubevirt/kubevirt/issues/9801#issuecomment-1570321772",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
***@***.***": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Comment by xpivarc** [comment]
Hi @caijian76 , please verify you have the balloon driver installed.

**Comment by caijian76** [comment]
> hello @caijian76, I am able to see the referred metrics for windows VMs, p.e.:
> 
> ```
> kubevirt_vmi_memory_used_bytes virt-handler metrics ... medium	windows-jvilaca kubevirt-prometheus-metrics	1554788352
> ```
> 
> what is the output of the `virsh dommemstat` in your VM virt-laucher pod?
> 
> ```
> > kubectl exec virt-launcher-<POD_SUFFIX> -- virsh list
>  Id   Name                      State
> -----------------------------------------
>  1    <DOMAIN>   running
> 
> > kubectl exec virt-launcher-<POD_SUFFIX> -- virsh dommemstat <DOMAIN>
> 
> actual 4194304
> swap_in 7440888
> swap_out 160388
> major_fault 272998
> minor_fault 40718945
> unused 2627208
> available 4174256
> usable 2655348
> last_update 1686315718
> disk_caches 56280
> hugetlb_pgalloc 0
> hugetlb_pgfail 0
> rss 4185564
> ```



 Id   Name          State
-----------------------------
 1    default_win   running

virsh # dommemstat default_win
actual 8388608
last_update 1686251422
rss 8502884

**Comment by caijian76** [comment]
> balloon

I have installed balloon driver in windows .name "VirtIO Bollon Driver"

**Comment by caijian76** [comment]
@machadovilaca No. I intall balloon driver whene Install windows.

**Comment by caijian76** [comment]
Run virsh ,I see `Authorization not available. Check if polkit service is running or see debug message for more information.`warn .Is it related to this?

**Comment by caijian76** [comment]
@machadovilaca 

For linux VM,that's right.
```
bash-5.1$ virsh list
Authorization not available. Check if polkit service is running or see debug message for more information.
 Id   Name                State
-----------------------------------
 1    default_ubuntu-10   running

bash-5.1$ virsh dommemstat default_ubuntu-10
Authorization not available. Check if polkit service is running or see debug message for more information.
actual 4194304
swap_in 0
swap_out 0
major_fault 1980
minor_fault 1207464
unused 2652888
available 4026068
usable 3616956
last_update 1686331038
disk_caches 1136196
hugetlb_pgalloc 0
hugetlb_pgfail 0
rss 2048444

bash-5.1$
```

**Comment by caijian76** [comment]
need blnsvr.ex -i install ballon server
